tests/unit/s2n_tls13_key_schedule_rfc8448_test.c

```diff
@@ -0,0 +1,381 @@
+
+                EXPECT_OK(s2n_tls13_key_schedule_update(conn));
+                EXPECT_BYTEARRAY_EQUAL(conn->secure.client_implicit_iv, iv.data, iv.size);
+            }
+        }
+    }
+
+    /* Resumed 0-RTT Handshake */
+    {
+        const uint32_t resumed_handshake_type = NEGOTIATED | WITH_EARLY_DATA;
+        const int resumed_message_nums[] = {
+                [CLIENT_HELLO] = 0,
+                [SERVER_FINISHED] = 3,
+        };
+
+        /**
+         *= https://www.rfc-editor.org/rfc/rfc8448.html#section-4
+         *#    {server}  extract secret "early" (same as client early secret)
+         *
+         *= https://www.rfc-editor.org/rfc/rfc8448.html#section-4
+         *#    {client}  extract secret "early":
+         *#
+         *#       salt:  0 (all zero octets)
+         *#
+         *#       IKM (32 octets):  4e cd 0e b6 ec 3b 4d 87 f5 d6 02 8f 92 2c a4 c5
+         *#          85 1a 27 7f d4 13 11 c9 e6 2d 2c 94 92 e1 c4 f3
+         *#
+         *#       secret (32 octets):  9b 21 88 e9 b2 fc 6d 64 d7 1d c3 29 90 0e 20
+         *#          bb 41 91 50 00 f6 78 aa 83 9c bb 79 7c b7 d8 33 2c
+         */
+        S2N_BLOB_FROM_HEX(early_secret, "9b 21 88 e9 b2 fc 6d 64 d7 1d c3 29 90 0e 20 \
+                     bb 41 91 50 00 f6 78 aa 83 9c bb 79 7c b7 d8 33 2c");
+
+        /**
+         *= https://www.rfc-editor.org/rfc/rfc8448.html#section-4
+         *#       hash (32 octets):  08 ad 0f a0 5d 7c 72 33 b1 77 5b a2 ff 9f 4c 5b
+         *#          8b 59 27 6b 7f 22 7f 13 a9 76 24 5f 5d 96 09 13
+         */
+        S2N_BLOB_FROM_HEX(client_hello_hash, "08 ad 0f a0 5d 7c 72 33 b1 77 5b a2 ff 9f 4c 5b \
+                     8b 59 27 6b 7f 22 7f 13 a9 76 24 5f 5d 96 09 13");
+
+        /* Derive early application traffic keys */
+        {
+            /**
+             *= https://www.rfc-editor.org/rfc/rfc8448.html#section-4
+             *#    {server}  derive read traffic keys for early application data (same
+             *#       as client early application data write traffic keys)
+             *
+             *= https://www.rfc-editor.org/rfc/rfc8448.html#section-4
+             *#    {client}  derive write traffic keys for early application data:
+             *#
+             *#       PRK (32 octets):  3f bb e6 a6 0d eb 66 c3 0a 32 79 5a ba 0e ff 7e
+             *#          aa 10 10 55 86 e7 be 5c 09 67 8d 63 b6 ca ab 62
+             *#
+             *#       key info (13 octets):  00 10 09 74 6c 73 31 33 20 6b 65 79 00
+             *#
+             *#       key expanded (16 octets):  92 02 05 a5 b7 bf 21 15 e6 fc 5c 29 42
+             *#          83 4f 54
+             *#
+             *#       iv info (12 octets):  00 0c 08 74 6c 73 31 33 20 69 76 00
+             *#
+             *#       iv expanded (12 octets):  6d 47 5f 09 93 c8 e5 64 61 0d b2 b9
+             */
+            S2N_BLOB_FROM_HEX(key, "92 02 05 a5 b7 bf 21 15 e6 fc 5c 29 42 \
```

Comment:
Okay I could be reading this test wrong, but where are you using these key blobs that get created?

Reply:
I see the `iv` being used, but I don't see any references to the `key`